src/main/java/com/flansmod/common/guns/EntityBullet.java

```diff
@@ -534,11 +541,14 @@ else if (motion.z != 0F)
                 Entity entity = (Entity) obj;
                 if (entity != this && entity != owner && !entity.isDead && !(entity instanceof EntityItem) && !(entity instanceof EntityXPOrb) && !(entity instanceof EntityArrow) &&
                         (!entity.getClass().toString().contains("flansmod.") || entity instanceof EntityAAGun || entity instanceof EntityGrenade)
-                        && !entity.getClass().toString().contains("holographicdisplays") && !entity.getClass().toString().contains("EntityScent")) {
-                    AxisAlignedBB bb = entity.boundingBox.addCoord(
+                        && !entity.getClass().toString().contains("holographicdisplays") && !entity.getClass().toString().contains("EntityScent")
+                        && !(entity instanceof EntityLivingBase && ((EntityLivingBase)entity).getHealth() == 0.0)) {
```

Comment:
When a mob is killed and in its "dying animation", its hitbox still catches bullets, even though the mob is flat on its side. That's a fix for that (checking if the entity is even alive)

Reply:
Hmm.. I'm not sure about that. Thanks for explaining though :)

Reply:
Yeah, getting class names using `entity.getClass().toString()` doesn't sit well with me either. Are these entity classes not in Flans/Forge?

Reply:
`holographicdisplays` comes from a bukkit plugin, and `EntityScent` is from a zombie AI mod thingy - so we can't check their classes directly. I was planning to make this dynamic against a list of names: https://github.com/Unknown025/Flans-Mod-Plus/issues/170

Idk if there's a better way to do this, if there's some kind of name thing.

Reply:
You could use `Class#forName(String)` to see if the classes exist at load time.

Reply:
That seems a similar level of jank to me, honestly. And I'm not sure how that'd help. Unless you want to skip the checks if they're not present? The aim is to allow bullets to pass through those entities, not get any of their properties - so we don't really need the classes.

Reply:
Well, it's a native method in Java to test if that class exists, and it's how SQL drivers are loaded, so I'd certainly consider it less jank. It'd help by giving an actual instance of a class to use for comparison, instead of just checking if a class contains a substring (or alternatively throwing a `ClassNotFoundException`).

Reply:
I see what you mean, yeah. But again it'd need to have all of these entity classes as dependencies on the mod at compile time (while keeping things simple), which cannot be dynamic.

I did some poking around and I agree, it's jankier than I thought. The .toString method of the class `entity.getClass().toString()` (not of the class, but of the class's class...er) gives `class net.minecraft.entity.monster.EntityCreeper`, but that assumes .toString is going to give something reasonable. And, it's messy.

Alternatively, `entity.toString()` prints out `EntityCreeper['Creeper'/179, l='MpServer', x=-376.69, y=71.10, z=-270.03]` which is `this.getClass().getSimpleName()` and `this.getCommandSenderName()` which seem a bit nicer to deal with - even if it's not a proper ID. I can try building a test out of these and seeing what happens.